### DISCUSSION TITLE
Running Dbus as System bus is failing - Ubuntu 20.0.4

### DISCUSSION THREAD

**Discussion by lazynomad** [AsamK/signal-cli#690] [Q&A]
Hey folks.. Been trying for 2 days with no luck... 
Trying to get Dbus run on system bus so I can have my another application (.net based if it matters) issue Send messages.

However I have been stuck with various issues.
Guide followed: [https://github.com/AsamK/signal-cli/wiki/DBus-service](url)

- I do not see signal.service file in repo. I assumed it could be "signal-cli.service" or "signal-cli@.service". Tried both but I made a little progress with the former.
- The "sed" command has ""s|%number%|<INSERT_YOUR_NUMBER>|"". However the file "signal-cli.service" DOES NOT have %number% parameter. So I don't know where is the "Sender" number provided as the dbus-send command has only Receipient number.
- I had to add "-u %I" to signal-cli.service file to make a little progress.
- I created a user "signal-cli" as per doc, but not sure what does it mean by "If you registered with user signal-cli, remove the config option.".. I removed the "--config" option anyway but uncertain if I missed some registration step.
- Once I get everything "somehow" running error free, then when I run the below command for first time, it doesn't work.

**Command**: dbus-send --system --type=method_call --print-reply --dest="org.asamk.Signal" /org/asamk/Signal org.asamk.Signal.sendMessage string:"Test message" array:string: string:+1234567890

**Error**: Error org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

Note: I tried restarting several times with no luck. I must be missing something very basic. Any help is appreciated.

PS: I do not have any issues running "signal-cli" commands directly. Dbus is what's failing for me.

**Comment by lazynomad** [comment]
New findings:

Figured the guide doesn't have the command "systemctl start signal-cli.service". However, when I run that, I get below error.
It indicates that the user number is not registered. However I can use signal-cli on the same machine to send messages. Should I register again for Dbus setup ? 

`signal-cli.service - Send secure messages to Signal clients
     Loaded: loaded (/etc/systemd/system/signal-cli.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Tue 2021-05-04 08:23:26 UTC; 19s ago
    Process: 7364 ExecStart=/opt/signal-cli-0.8.1/bin/signal-cli -u +1987654321 daemon --system (code=exited, status=1/FAILURE)
   Main PID: 7364 (code=exited, status=1/FAILURE)

May 04 08:23:25 systemd[1]: Starting Send secure messages to Signal clients...
May 04 08:23:26 signal-cli[7364]: User +1987654321 is not registered.
May 04 08:23:26 systemd[1]: signal-cli.service: Main process exited, code=exited, status=1/FAILURE
May 04 08:23:26 systemd[1]: signal-cli.service: Failed with result 'exit-code'.
May 04 08:23:26 systemd[1]: Failed to start Send secure messages to Signal clients.
`

**Comment by zimtkeks** [comment]
I had a similar error about user being not registered.
In my case, the `signal-cli` instance run by systemd was unaware of [config files](https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#files) containing the user info.
You can use the `--config` parameter to fix that, but be aware that file permissions should allow access by systemd-`signal-cli`. At the same time, I suppose access by other users on the system should be prevented.

**Comment by riazm** [comment]
The wiki instructions are out of date for sure, at the very least they need to be updated. Ideally with instructions how to remove the old signal.service files.

I personally get an error like:

`2021-06-04T14:18:23.166Z [main] DEBUG org.freedesktop.dbus.connections.transports.UnixSocketTransport - Disconnecting Transport
2021-06-04T14:18:23.166Z [main] DEBUG org.freedesktop.dbus.spi.OutputStreamMessageWriter - Closing Message Writer
Failed to send message: Unit signal-cli.service is not loaded properly: File exists.`
